lib/backup/notifier/command.rb

```diff
@@ -0,0 +1,99 @@
+# encoding: utf-8
+
+module Backup
+  module Notifier
+    class Command < Base
+
+      ##
+      # Command to execute.
+      #
+      # Make sure it is accessible from your $PATH, or provide
+      # the absolute path to the command.
+      attr_accessor :command
+
+      ##
+      # Arguments to pass to the command.
+      #
+      # Must be an array of strings or callable objects.
+      #
+      # Callables will be invoked with #call(model, status),
+      # and the return value used as the argument.
+      #
+      # In strings you can use the following placeholders:
+      #
+      # %l - Model label
+      # %t - Model trigger
+      # %s - Status (success/failure/warning)
+      # %v - Status verb (succeeded/failed/succeeded with warnings)
+      #
+      # All placeholders can be used with uppercase letters to capitalize
+      # the value.
+      #
+      # Defaults to ["%L %v"]
+      attr_accessor :args
+
+      def initialize(model, &block)
+        super
+        instance_eval(&block) if block_given?
+
+        @args ||= ["%L %v"]
+      end
+
+      private
+
+      ##
+      # Notify the user of the backup operation results.
+      #
+      # `status` indicates one of the following:
+      #
+      # `:success`
+      # : The backup completed successfully.
+      # : Notification will be sent if `on_success` is `true`.
+      #
+      # `:warning`
+      # : The backup completed successfully, but warnings were logged.
+      # : Notification will be sent if `on_warning` or `on_success` is `true`.
+      #
+      # `:failure`
+      # : The backup operation failed.
+      # : Notification will be sent if `on_warning` or `on_success` is `true`.
+      #
+      def notify!(status)
+        IO.popen([@command] + args.map { |arg| format_arg(arg, status) })
```

Comment:
You can add ENV vars by adding them to the start of the command.

```ruby
IO.popen("FOO=bar ruby ./bla.rb --foo=bar FOO2=baz")
```

FOO will be regarded as an ENV var, but FOO2 is indeed an argument. As long as the command can be completely customized it shouldn't be a problem.

What I mean with not building a complete custom command is that so far I haven't done the string replacement you added in the `format_arg` method. I mean that you can specify what command to run `my-executable --my-arg=my-val`, but that no string replacement is done. Instead all the data that is available from backup is sent with ENV variables. So `BACKUP_PROCEDURE=my_application` and `BACKUP_STATUS=failure`. And my question was if you have any preference over string replacement or a bunch of ENV vars at all times.